### PR TITLE
[WIP] Bulk memory operations

### PR DESCRIPTION
**Pull request by tlively**
The only parts missing are binaryen.js test results, spec tests, and
the interpreter implementation. The fuzzer has also not been
successfully run.

### CODE REVIEW

File: src/binaryen-c.h [hunk 1]
```diff
@@ -122,6 +122,10 @@ BinaryenExpressionId BinaryenSIMDReplaceId(void);
 BinaryenExpressionId BinaryenSIMDShuffleId(void);
 BinaryenExpressionId BinaryenSIMDBitselectId(void);
 BinaryenExpressionId BinaryenSIMDShiftId(void);
+BinaryenExpressionId BinaryenMemoryCopyId(void);
+BinaryenExpressionId BinaryenDataDropId(void);
+BinaryenExpressionId BinaryenMemoryCopyId(void);
```
Comment (kripken) [inline src/binaryen-c.h]: `BinaryenMemoryCopyId` appears to be duplicated?

Comment (MaxGraey) [reply src/binaryen-c.h]: It seems this should be `BinaryenMemoryInitId`

File: src/wasm-stack.h [hunk 2]
```diff
@@ -954,6 +958,47 @@ void StackWriter<Mode, Parent>::visitSIMDShift(SIMDShift* curr) {
...
 }
 
+template<StackWriterMode Mode, typename Parent>
+void StackWriter<Mode, Parent>::visitMemoryInit(MemoryInit* curr) {
+  visitChild(curr->dest);
+  visitChild(curr->offset);
+  visitChild(curr->size);
+  if (justAddToStack(curr)) return;
+  o << int8_t(BinaryConsts::TruncSatPrefix);
+  o << U32LEB(BinaryConsts::MemoryInit);
```
Comment (kripken) [inline src/wasm-stack.h]: Do the bulk memory operatoins use the trunc_sat prefix?

Comment (tlively) [reply src/wasm-stack.h]: They do. This prefix should probably be renamed to `MiscPrefix`, as it is called in the spec proposals.

Comment (kripken) [reply src/wasm-stack.h]: I see, thanks. Yeah, let's rename to that then, looks odd otherwise.

**Pull request merged**